Add project description and boolean functions

gschult opened the pull request:
@kirschmet2 If you can think of any code or documentation improvements, please add comments. Especially point out if you fail to understand something.

gschult committed:
mob next [ci-skip] [ci skip] [skip ci]

lastFile:app/Main.hs

gschult committed:
Implement boolean functions

Implemented xor using == because it's easier to do. One could try to
completely eliminate the Prelude to make the code more self sufficient.

gschult committed:
Add project description

kirschmet2 commented on app/Main.hs:
```diff
@@ -1,8 +1,10 @@
+import Prelude (IO, print)
+
+import Bits
 
 main :: IO ()
-main = do
-  putStrLn "Hello, Haskell!"
-  MyLib.someFunc
+main = print (nor I I)
```
``` suggestion
import Prelude (IO, print, (==))

import Bits

main :: IO ()
main = print (nor I I == O) -- should print true
```
Expected output of `(nor I I)` was not obvious, made it explicit. This also demonstrates that boolean logic can be applied to our custom `Bit` type.

gschult replied:
Good idea. Thanks.

gschult committed:
Make expected output more explicit

Expected output of (nor I I) was not obvious, made it explicit.
This also demonstrates that boolean logic can be applied to our custom Bit type.
